packages/next/types/index.d.ts

```diff
@@ -80,6 +85,7 @@ export type GetStaticPropsContext<Q extends ParsedUrlQuery = ParsedUrlQuery> = {
 
 export type GetStaticPropsResult<P> = {
   props: P
+  unstable_redirect?: Redirect
```

Comment:
The type is incorrect, you can either return `unstable_redirect` or `props`, not both